### PR TITLE
do not claim any disk that has non-Cinder role assigned

### PR DESCRIPTION
**Pull request by jsuchome**
based on https://github.com/crowbar/barclamp-crowbar/pull/1306 and https://github.com/crowbar/barclamp-deployer/pull/248

### CODE REVIEW

File: chef/cookbooks/cinder/recipes/volume.rb [hunk 1]
```diff
@@ -88,8 +88,16 @@ def make_volume(node, backend_id, volume)
 
   return if volume_exists(volname)
 
-  unclaimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.unclaimed(node)
-  claimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.claimed(node, "Cinder")
+  claim_string = "Cinder"
```
Comment (tboerger) [inline chef/cookbooks/cinder/recipes/volume.rb]: I would prefer snakecase claim strings as we anyway display some I18n key within crowbar.

Comment (tboerger) [inline chef/cookbooks/cinder/recipes/volume.rb]: Ok, forget it. we are already using this name.

**Comment by tboerger** [comment]
LGTM

File: chef/cookbooks/cinder/recipes/volume.rb [hunk 2]
```diff
@@ -88,8 +88,16 @@ def make_volume(node, backend_id, volume)
...
+
+  # do not claim any disk that has non-Cinder role assigned
+  disk_roles = node["crowbar_wall"].disk_roles || {}
+  unclaimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.unclaimed(node).sort.select { |d|
+    role = disk_roles[d.unique_name] || ""
+    role.empty? || role == claim_string
```
Comment (dirkmueller) [inline chef/cookbooks/cinder/recipes/volume.rb]: shouldn't unclaimed() return only disks with role.empty? shouldn't this be fixed in the unclaimed method itself?

Comment (jsuchome) [inline chef/cookbooks/cinder/recipes/volume.rb]: Well, it is still something different. Disk could have role assigned, but is still unclaimed until it is explicitely claimed

Comment (jsuchome) [inline chef/cookbooks/cinder/recipes/volume.rb]: > shouldn't unclaimed() return only disks with role.empty?

We really want the disks that are unclaimed AND with Ceph role. Or with empty role, to simulate the old behavior when user doesn't have roles set

Comment (vuntz) [inline chef/cookbooks/cinder/recipes/volume.rb]: @jsuchome but it does make sense to add some helper in the barclamp library to do what you're doing here (and in the other barclamps that use this disk role stuff), doesn't it?

Comment (jsuchome) [inline chef/cookbooks/cinder/recipes/volume.rb]: @vuntz @dirkmueller addressed in https://github.com/crowbar/barclamp-deployer/pull/248

**Comment by tboerger** [comment]
:+1:

**Comment by dirkmueller** [comment]
+1

**Comment by vuntz** [comment]
+1, but I understand code might change after discussion in https://github.com/crowbar/barclamp-deployer/pull/248

**Comment by rsalevsky** [comment]
@jsuchome, this repository got merged into [crowbar-openstack](https://github.com/crowbar/crowbar-openstack).
Please open a new pull request there. __Do NOT merge here anymore!__

**Comment by jsuchome** [comment]
Closing, as there's already PR to new repo

**Pull request closed without merging**